File: DSA Practice/LeetCode/260. Single Number III/solution.java
class Solution {
    public int[] singleNumber(int[] nums) {
        HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();

        for (int i : nums){
            if (m.containsKey(i)){
                m.put(i, m.get(i)+1);
            } else {
                m.put(i, 1);
            }
        }

        int[] res = new int[2];
        int index = 0;
        for (Map.Entry i : m.entrySet()){
            if ((int) i.getValue() == 1){
                res[index] = (int) i.getKey();
                index++;
            }
        }

        return res;
    }
}

File: DSA Practice/LeetCode/260. Single Number III/solution.kt
class Solution {
    fun singleNumber(nums: IntArray): IntArray {
        var m : HashMap<Int, Int?> = HashMap<Int, Int?>()

        for (i in nums){
            if (m.containsKey(i)){
                m.put(i, m.get(i)?.plus(1))
            } else {
                m.put(i, 1)
            }
        }

        var res : IntArray = IntArray(2)
        var index : Int = 0

        for (i in m.keys){
            if (m.get(i) == 1){
                res.set(index, i)
                index++
            }
        }

        return res
    }
}